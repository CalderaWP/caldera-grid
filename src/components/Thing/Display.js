import React from 'react';
import {displayPropTypes} from "./PropTypes";

export const  Display = (props) => {
    const thing = undefined !== props.thing ? props.thing : 'Nothing';
    return (
        <div>{thing}</div>
    );
};


Display.propTypes = displayPropTypes;

