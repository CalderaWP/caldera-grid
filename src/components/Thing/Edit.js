import React from 'react';
import {editPropTypes} from "./PropTypes";

export const Edit = (props) => {
    return (
        <div>
            <label
                htmlFor={props.idAttr}
            >
                {props.label}
            </label>
            <input
                id={props.idAttr}
                value={props.thing}
                onChange={(event) => {
                    props.onChange(event.target.value);
                }}
            />
        </div>
    );
};

Edit.propTypes = editPropTypes;