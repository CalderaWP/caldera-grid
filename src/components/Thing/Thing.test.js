import React from 'react';
import {Display} from "./Display";
import {Edit} from "./Edit";
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import { mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

class InputBox extends React.Component {
    onSearch(event) {
        event.preventDefault();
        this.props.onSearch(event.target.value.trim());
    }
    render () { return <input onChange={this.onSearch.bind(this)} />; }
}

it('should call onChange prop', () => {
    const newValue = 'Mike R3S Corkum';
    const onChangeMock = jest.fn();
    const event = {
        preventDefault() {},
        target: { value: newValue }
    };
    const idAttr = 'thing-chooser';
    const component = mount(<Edit
        onChange={onChangeMock}
        idAttr={idAttr}
        label={'Choose Thing'}
        thing={'A'}
    />);

    expect(component.prop('idAttr')).toBe(idAttr);
    expect(component.prop('thing')).toBe('A');

    component.find('input').simulate('change', event);
    expect(onChangeMock).toBeCalledWith(newValue);
    component.setProps({ thing: newValue });
    expect(component.prop('thing')).toBe(newValue);
});


configure({ adapter: new Adapter() });
test('Test default', () => {
    const component = renderer.create(
        <Display/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Test with message ', () => {
    const component = renderer.create(
        <Display thing={'Hi Roy'}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Test with message ', () => {
    const component = renderer.create(
        <Edit
            thing={'Hi Roy'}
            idAttr={'thing-chooser'}
            label={'Change Thing'}
            onChange={(event) => {}}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

