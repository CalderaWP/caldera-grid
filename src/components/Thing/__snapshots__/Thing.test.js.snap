// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test default 1`] = `
<div>
  Nothing
</div>
`;

exports[`Test with message  1`] = `
<div>
  Hi Roy
</div>
`;

exports[`Test with message  2`] = `
<div>
  <label
    htmlFor="thing-chooser"
  >
    Change Thing
  </label>
  <input
    id="thing-chooser"
    onChange={[Function]}
    value="Hi Roy"
  />
</div>
`;
